### PR TITLE
kismet: 2022-08 -> 2023-07. disable coconut.

### PR DESCRIPTION
**Pull request by hughobrien**
## Description of changes

Update to kismet, for full details please see the [release notes](https://www.kismetwireless.net/posts/kismet-2023-07-r1/)

Note: This does not build support for the [WiFi Coconut](https://shop.hak5.org/products/wifi-coconut) as there is currently an [open blocker](https://github.com/kismetwireless/kismet/issues/478).

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by sarcasticadmin — CHANGES_REQUESTED** [review]
@hughobrien this makes sense give the current limitation with the wifi coconut. I can confirm that its indeed disabled now from the build log:

```
kismet>             ADSB Proxy: yes
kismet>          bladeRF-wiphy: no (bladeRF support not enabled)
kismet>      Hak5 WiFi Coconut: no (Hak5 WiFi Coconut support not enabled)
kismet>  lm-sensors monitoring: no (will not be able to monitor system temperature, etc)
kismet>         Built-in Debug: yes - Full debug info available on crash, using unwind
```

Would you mind splitting up this commit into 2 commits? One for the version bump and the other for the configure flag so its aligns with the contributing guidelines: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

**Comment by sarcasticadmin** [comment]
Result of `nixpkgs-review pr 262454` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kismet</li>
  </ul>
</details>

**Review by sarcasticadmin — APPROVED** [review]

File: pkgs/applications/networking/sniffers/kismet/default.nix [hunk 1]
```diff
@@ -87,6 +87,7 @@ stdenv.mkDerivation rec {
   ];
 
   configureFlags = [
+  "--disable-wifi-coconut"  # Until https://github.com/kismetwireless/kismet/issues/478
```
Comment (GetPsyched) [inline pkgs/applications/networking/sniffers/kismet/default.nix:90]: Minor indentation issue

Comment (hughobrien) [reply pkgs/applications/networking/sniffers/kismet/default.nix]: good catch. fixed.

**Review by pbsds — APPROVED** [review]
Considering that coconut was not a part of the prior release (it was added august 2022?), and their release cadence not being any real indication of this being a major release, i only see their switch to JWT tokens as the only technical breaking change here, which is fairly transparent to the user. Should be fine to merge for 23.11

Result of `nixpkgs-review pr 262454` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kismet</li>
  </ul>
</details>

Tested a little in the webui. LGTM

Some files in `share/kismet/httpd/css/` are marked executable, could this be fixed?

**Comment by hughobrien** [comment]
The exec issue is [upstream](https://github.com/kismetwireless/kismet/blob/master/http_data/css/reset.css) - I'll raise an issue.

**Comment by hughobrien** [comment]
upstream fixed merged https://github.com/kismetwireless/kismet/pull/489/files

**Pull request merged**